microbench/src/main/java/io/netty/microbench/buffer/SimpleByteBufPooledAllocatorBenchmark.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.microbench.buffer;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.buffer.PooledByteBufAllocator;
+import io.netty.microbench.util.AbstractMicrobenchmark;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.infra.Blackhole;
+
+import java.util.concurrent.TimeUnit;
+
+@State(Scope.Benchmark)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+public class SimpleByteBufPooledAllocatorBenchmark extends AbstractMicrobenchmark {
+
+    public SimpleByteBufPooledAllocatorBenchmark() {
+        super(true, false);
+    }
+
+    @Param({"123", "1234", "12345", "123456", "1234567"})
+    public int size;
+
+    @Param({"0", "5", "10", "100"})
+    public long tokens;
+
+    @Benchmark
+    public boolean getAndRelease() {
+        ByteBufAllocator alloc = PooledByteBufAllocator.DEFAULT;
```

Comment:
@netudima can you create a new `PooledByteBufAllocator` instance in a `@Setup` method and use it ? Otherwise it may not produce fair results as it may have already stuff cached from before etc. You may even want to disable `ThreadLocal` caches all together when doing so to ensure you really hit the `PoolArena`. WDYT ?

Reply:
@franz1981 WDYT ?

Reply:
I missed that :O ,that's a good point :+1

Reply:
@netudima please address and let me know once done... after all my comments are addressed I will merge this. Good job.

Reply:
Thank you for the review, working on the changes.

Reply:
Added
```
@Param({"false", "true"})
    public boolean useThreadCache;
```
to control thread cache usage in the bench